### PR TITLE
fix: differentiate between passed and rejected proposals

### PR DESCRIPTION
**Pull request by DanThomp507**
This also includes resolving the following bug: 

Update executeApprove and executeReject to verify if signer is sub pod member

### CODE REVIEW

File: src/Proposal.ts [hunk 1]
```diff
@@ -86,9 +87,10 @@ export default class Proposal {
       : this.pod.threshold;
     this.safeTransaction = safeTransaction;
     this.rejectTransaction = rejectTransaction;
+    this.status = status;
 
     if (podNonce === this.id) this.status = 'active';
-    if (podNonce > this.id) this.status = 'executed';
+    if (podNonce > this.id) this.status = 'passed';
```
Comment (0xGrapefruit) [inline src/Proposal.ts:93]: We can leave this as-is but we should probably differentiate between executed proposals that were passed vs rejected. Can you add a TODO here? I'll make a ticket once you do.

File: test/proposal.test.ts [hunk 2]
```diff
@@ -111,11 +111,11 @@ describe('Pod.getProposals', () => {
     );
   });
 
-  test('Pod.getProposals({ status: executed }) will return only the active proposal', async () => {
+  test('Pod.getProposals({ status: passed || rejected }) will return only the active proposal', async () => {
```
Comment (0xGrapefruit) [inline test/proposal.test.ts:114]: I don't think this should be the functionality. This is also a breaking change. I think we can leave the status as 'executed' and it still returns all passed and rejected proposals. Thoughts? @jtlacci

File: src/Proposal.ts [hunk 3]
```diff
@@ -86,9 +87,11 @@ export default class Proposal {
       : this.pod.threshold;
     this.safeTransaction = safeTransaction;
     this.rejectTransaction = rejectTransaction;
+    this.status = status;
 
-    if (podNonce === this.id) this.status = 'active';
-    if (podNonce > this.id) this.status = 'executed';
+    // TODO: differentiate between passed and executed proposals
+    if (podNonce === this.id) this.status = 'executed';
```
Comment (0xGrapefruit) [inline src/Proposal.ts:93]: I don't think this is correct, it should still be 'active' if the nonce === the proposal ID. I think you can keep line 94 as is though.

File: src/Pod.ts [hunk 4]
```diff
@@ -282,10 +287,18 @@ export default class Pod {
       const rejectNonceIndex = rejectNonces.indexOf(tx.nonce);
       // If there is, we package that together with the regular transaction.
       if (rejectNonceIndex >= 0) {
-        return new Proposal(this, nonce, tx, rejectTransactions[rejectNonceIndex]);
+        // Check to see if the reject transaction has been executed
```
Comment (0xGrapefruit) [inline src/Pod.ts:290]: Is it possible to move this logic into the Proposal class, I'd prefer not to change the Proposal constructor signature if possible.

File: src/Pod.ts [hunk 5]
```diff
@@ -217,7 +217,11 @@ export default class Pod {
       case 'active':
         params = { nonce };
         break;
-      case 'executed':
+      case 'passed':
```
Comment (0xGrapefruit) [inline src/Pod.ts:220]: Can you keep the 'executed' case for now? I think we might actually want to remove the 'passed' and 'rejected' cases for now as well

File: test/proposal.test.ts [hunk 6]
```diff
@@ -157,7 +157,7 @@ describe('Proposal details', () => {
     const proposal = (await pod.getProposals())[0];
 
     expect(proposal.id).toBe(1);
-    expect(proposal.status).toBe('active');
+    expect(proposal.status).toBe('executed');
```
Comment (0xGrapefruit) [inline test/proposal.test.ts:160]: I don't think this should have changed either, but I think one of my earlier comments will address this

File: test/proposal.test.ts (hunk 2 above)
Comment (jtlacci) [reply test/proposal.test.ts]: i agree, also would prefer we didn't introduce breaking changes

**Review by 0xGrapefruit — APPROVED** [review]

**Pull request merged**